Corona-Warn-App/src/test/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/exportAll/DccExportAllOverviewViewModelTest.kt

```diff
@@ -0,0 +1,131 @@
+
+        val rc = mockk<RecoveryCertificate>().apply {
+            every { fullNameFormatted } returns "Full Name"
+            every { dateOfBirthFormatted } returns "1990-10-10"
+            every { uniqueCertificateIdentifier } returns "UNIQUE_CERTIFICATE_IDENTIFIER"
+            every { targetDisease } returns "Covid 19"
+            every { certificateCountry } returns "Germany"
+            every { qrCodeBase64() } returns "1234"
+            every { certificateIssuer } returns "Robert Koch"
+            every { validFromFormatted } returns "2020-12-10"
+            every { validUntilFormatted } returns "2021-12-10"
+            every { testedPositiveOnFormatted } returns "2021-11-10"
+            every { state } returns CwaCovidCertificate.State.Valid(org.joda.time.Instant.now())
+            every { validFrom } returns LocalDate.now()
+            every { validUntil } returns LocalDate.now()
+            every { fullName } returns "A"
+            every { testedPositiveOn } returns LocalDate.now()
+            every { rawCertificate.payload.certificateCountry } returns "DE"
+        }
+
+        val tc = mockk<TestCertificate>().apply {
+            every { fullNameFormatted } returns "Full Name"
+            every { dateOfBirthFormatted } returns "1990-10-10"
+            every { uniqueCertificateIdentifier } returns "UNIQUE_CERTIFICATE_IDENTIFIER"
+            every { targetDisease } returns "Covid 19"
+            every { certificateCountry } returns "Germany"
+            every { qrCodeBase64() } returns "1234"
+            every { certificateIssuer } returns "Robert Koch"
+            every { testName } returns "Rapid"
+            every { testNameAndManufacturer } returns "Robert Koch"
+            every { sampleCollectedAtFormatted } returns "2020-10-20"
+            every { testResult } returns "Negative"
+            every { testCenter } returns "TestIQ"
+            every { testType } returns "Rapid"
+            every { state } returns CwaCovidCertificate.State.Valid(org.joda.time.Instant.now())
+            every { sampleCollectedAt } returns org.joda.time.Instant.now()
+            every { fullName } returns "A"
+            every { rawCertificate.payload.certificateCountry } returns "DE"
+        }
+        every { template.invoke(any()) } returns "Template"
+        every { personCertificatesProvider.certificateContainer } returns flowOf(
+            mockk<CertificateProvider.CertificateContainer>().apply {
+                every { allCwaCertificates } returns setOf(vc, rc, tc)
+            }
+        )
+        every { timeStamper.nowJavaUTC } returns Instant.now()
+    }
+
+    @Test
+    fun getPdfString() {
+        instance().pdfString.getOrAwaitValue() shouldBe HTML_TEMPLATE.replace(
+            oldValue = "++certificates++",
+            newValue = """
+                <li>Template</li>
+                <li>Template</li>
+                <li>Template</li>
+            """.trimIndent()
+        )
+    }
+
```

Comment:
You could also write a test for sharePDF()

Reply:
will try in the other PR , not so easy to unit test Android APIs in unit tests 🤷‍♂️